src/hardware/ipx.cpp

```diff
@@ -592,23 +605,63 @@ static void receivePacket(uint8_t *buffer, int16_t bufSize) {
 	LOG_IPX("IPX: RX Packet loss!");
 }
 
-static void IPX_ClientLoop(void) {
-	int numrecv;
-	UDPpacket inPacket;
-	inPacket.data = (Uint8 *)recvBuffer;
-	inPacket.maxlen = IPXBUFFERSIZE;
-	inPacket.channel = UDPChannel;
+static void IPX_InitializePacketQueue()
+{
+	for (int i = 0; i < PacketQueueSize; i++) {
+		ipx_packet_queue[i].available = true;
+	}
+}
 
-	// Its amazing how much simpler UDP is than TCP
-	numrecv = SDLNet_UDP_Recv(ipxClientSocket, &inPacket);
-	if(numrecv) receivePacket(inPacket.data, inPacket.len);
+static void IPX_ClientLoop(void)
+{
+	UDPpacket in_packet;
+
+	for (int i = 0; i < PacketQueueSize; i++) {
+		if (!ipx_packet_queue[i].available) {
+			in_packet = ipx_packet_queue[i].udp;
+			receivePacket(in_packet.data, in_packet.len);
+			ipx_packet_queue[i].available = true;
+		}
+	}
 }
 
+void IPX_UdpClientListener()
```

Comment:
`UdpClientListener` -- that's a noun. What does this function do? Needs a verb somewhere 😄 

Edit: okay, a callback. Let's call it `udp_client_listener_callback` then. Make is static too as it's a private method.

We only use `MODULENAME_FunctionName` for the public API of a module, otherwise private functions should all be `lower_snake_case` and static.